web_apply_field_style/README.rst

```diff
@@ -0,0 +1,101 @@
+=================
+Apply Field Style
+=================
+
+.. 
+   !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+   !! This file is generated by oca-gen-addon-readme !!
+   !! changes will be overwritten.                   !!
+   !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+   !! source digest: sha256:dec2135f450935a31305575e98d577ccafcc5e10e314dc396f4e0c2f15f9e691
+   !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+.. |badge1| image:: https://img.shields.io/badge/maturity-Beta-yellow.png
+    :target: https://odoo-community.org/page/development-status
+    :alt: Beta
+.. |badge2| image:: https://img.shields.io/badge/licence-AGPL--3-blue.png
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+.. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fweb-lightgray.png?logo=github
+    :target: https://github.com/OCA/web/tree/16.0/web_apply_field_style
+    :alt: OCA/web
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/web-16-0/web-16-0-web_apply_field_style
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runboat-Try%20me-875A7B.png
+    :target: https://runboat.odoo-community.org/builds?repo=OCA/web&target_branch=16.0
+    :alt: Try me on Runboat
+
+|badge1| |badge2| |badge3| |badge4| |badge5|
+
+Allow to set an additional css class to fields in form view.
+
+Use case : you may highlight some fields for training purpose
+
+.. figure:: https://raw.githubusercontent.com/OCA/web/16.0/web_apply_field_style/static/description/demo.png
+    :alt: Colored fields
+
+**Table of contents**
+
+.. contents::
+   :local:
+
+Configuration
+=============
+
+Override _get_field_styles() with a dict of fields list per model
+
+
+.. code-block:: python
+
+    class Base(models.AbstractModel):
+        _inherit = "base"
+
+        def _get_field_styles(self):
+            res = super()._get_field_styles()
+            res["product.product"] = {
+                "my-css-class1": ["field1", "field2"],
+                "my-css-class2": ["field3"],
+            }
+            return res
```

Comment:
```suggestion
    class ProductProduct(models.Model):
        _inherit = "product.product"

        def _get_field_styles(self):
            return = {
                "field1": ["my-css-class1", "my-css-class2"],
                "field2": ["my-css-class1"],
            }
```
Not a blocking point, but I see this implementation more simple.
Of course, it requires some refactoring.

Reply:
Thanks for the suggestion.
On my side I would use that module in the global scope of a project. I would inherit this method at the level of my project.
Then it's easier for me to only call the method once instead of for every model

What is your use case to call the method for each model ?

Reply:
> What is your use case to call the method for each model ?

well, I'm not sure to use this module, but I guess, that the usage could be, when we develop a new module to add this function in the model we are creating / orverloading.
That case, overloading base doesn't make sense, and force to create a dedicated file.

but I understand your needs also.

Don't know !